recipe/gen_patch_json.py

```diff
@@ -239,6 +239,22 @@ def rename_dependency(fn, record, old_name, new_name):
             depends[dep_idx] = new_name + remainder
             instructions["packages"][fn]['depends'] = depends
 
+    def fix_libgfortran(fn, record):
+        depends = record.get("depends", ())
+        dep_idx = next(
+            (q for q, dep in enumerate(depends) if dep.split(' ')[0] == "libgfortran"),
+            None
+        )
+        if dep_idx:
+            # make sure respect minimum versions still there
+            # default to a min of 3.0.1 if none is given
+            if ">=3.0.1" in depends[dep_idx] or depends[dep_idx] == "libgfortran":
+                rename_dependency(fn, record, depends[dep_idx], "libgfortran >=3.0.1,<4.0.0.a0")
+            elif ">=4.0.0" in depends[dep_idx]:
+                rename_dependency(fn, record, depends[dep_idx], "libgfortran >=4.0.0,<5.0.0.a0")
+            else:
+                rename_dependency(fn, record, depends[dep_idx], "libgfortran >=3.0.0,<4.0.0.a0")
```

Comment:
This case should catch any other variants

Reply:
```
'libgfortran',
 'libgfortran ==3.0.1',
 'libgfortran >=3.0',
 'libgfortran >=3.0.1'
```

Here are the 4 variants I see in conda-forge packages. I think we don't want to change a specific pin like `libgfortran ==3.0.1` here.

Reply:
Ahh ok. Will fix.

Reply:
Done.